v1/client.go

```diff
@@ -147,6 +148,7 @@ func (c *Client) init() error {
 	}
 	t := cleanhttp.DefaultTransport()
 	t.TLSClientConfig = tlsConfig
+	t.ExpectContinueTimeout = 5 * time.Second
```

Comment:
Go docs are vague as to what happens after the timeout is reached. Does it send the body or abort the request?

In this case if we do not want / expect archivist to handle the upload body I'm curious if we should try to bail out / cancel the request if we don't get the redirect when we expect to.